openai/openai.go

```diff
@@ -175,18 +176,29 @@ func New(opts ...Option) (*Client, error) {
 	if cfg.proxyURL != "" {
 		proxy, _ := url.Parse(cfg.proxyURL)
 		httpClient.Transport = &http.Transport{
-			Proxy: http.ProxyURL(proxy),
+			Proxy:               http.ProxyURL(proxy),
+			MaxIdleConns:        100,
+			MaxIdleConnsPerHost: 100,
+			TLSHandshakeTimeout: 5 * time.Second,
 		}
 	} else if cfg.socksURL != "" {
 		dialer, err := proxy.SOCKS5("tcp", cfg.socksURL, nil, proxy.Direct)
 		if err != nil {
 			return nil, fmt.Errorf("can't connect to the proxy: %s", err)
 		}
 		httpClient.Transport = &http.Transport{
-			Dial: dialer.Dial,
+			Dial:                dialer.Dial,
+			MaxIdleConns:        100,
+			MaxIdleConnsPerHost: 100,
+			TLSHandshakeTimeout: 5 * time.Second,
+		}
+	} else {
+		httpClient.Transport = &http.Transport{
+			MaxIdleConns:        100,
+			MaxIdleConnsPerHost: 100,
+			TLSHandshakeTimeout: 5 * time.Second,
 		}
 	}
-
 	c.HTTPClient = httpClient
 	instance.client = openai.NewClientWithConfig(c)
```

Comment:
Overall, the code patch looks good and there are no obvious bug risks or security vulnerabilities. 

One minor improvement suggestion could be to handle errors more robustly, instead of just returning nil or printing an error message. For example, if there is an error connecting to the proxy or SOCKS5 server, it may be better to wrap the error and return it as part of the function's return value, so that the caller can handle the error appropriately. 

Additionally, the code could benefit from some comments or documentation to explain what each section of the code is doing and why certain options are being set. This can help other developers understand and maintain the code more easily.